core/src/main/java/org/infinispan/globalstate/impl/GlobalConfigurationManagerImpl.java

```diff
@@ -270,7 +272,20 @@ public CompletionStage<Configuration> getOrCreateCache(String cacheName, String
    }
 
    CompletionStage<Configuration> createCache(String cacheName, String template, Configuration configuration, EnumSet<CacheContainerAdmin.AdminFlag> flags) {
-      return createCacheInternal(cacheName, template, configuration, flags).thenApply((v) -> configuration);
+      CompletionStage<Configuration> cs = createCacheInternal(cacheName, template, configuration, flags)
+            .thenApply((v) -> configuration);
+      if (!CompletionStages.isCompletedSuccessfully(cs)) {
+         outstandingCreate.compute(cacheName, (ignore, existing) -> {
+            if (existing == null) return cs.thenRun(() -> outstandingCreate.remove(cacheName));
+
+            CompletableFuture<Void> o = existing.toCompletableFuture();
+            return cs.whenComplete((res, t) -> {
+               if (t != null) o.completeExceptionally(t);
```

Comment:
AFAICT the exception case is never handled, as `createCache` just returns the `cs` stage and in `createCacheLocally` we only ever call `thenCompose` on the returned stage. At the very least we should log the failure.